pkg/component/delete/delete.go

```diff
@@ -74,3 +85,95 @@ func references(list []unstructured.Unstructured, ownerRef metav1.OwnerReference
 	}
 	return false
 }
+
+// ListResourcesToDeleteFromDevfile parses all the devfile components and returns a list of resources that are present on the cluster and can be deleted
+func (do DeleteComponentClient) ListResourcesToDeleteFromDevfile(devfileObj parser.DevfileObj, appName string) (map[string]string, []unstructured.Unstructured, error) {
+	// resources will be passed on to DeleteComponentClient.DeleteResources to delete outerloop resources;
+	// it will only contain resources pushed to the cluster
+	var resources []unstructured.Unstructured
+	// innerLoopLabels will be passed on to DeleteComponentClient.DeleteDevfileComponent() to fetch the deployment for deletion;
+	// an empty list is returned if the deployment is not found on the cluster
+	var innerLoopLabels map[string]string
+
+	// Fetch the deployment of the devfile component
+	componentName := devfileObj.GetMetadataName()
+	deployment, err := do.kubeClient.GetDeploymentByName(fmt.Sprintf("%s-%s", componentName, appName))
+	if err == nil {
+		innerLoopLabels = deployment.Labels
```

Comment:
Instead of returning labels to be used to delete the deployment by labels, you should return the deployment as part of the resources list, so it is identified by its kind/name, and not by its labels anymore.

I think it is dangerous to identify the deployment to delete by its labels, as you are not 100% sure there won't have other deployment with the same labels.

Going this way, you would have to change `DeleteDevfileComponent` to be some kind of `PreDeleteDevfileComponent` which will run the hooks before to delete the component, but won't delete the deployment itself (as it will be deleted as part of the call to `DeleteResources`)

Reply:
>Instead of returning labels to be used to delete the deployment by labels, you should return the deployment as part of the resources list, so it is identified by its kind/name, and not by its labels anymore.

I initially had that thought, but I thought we didn't want to return a low layer k8s object to the CLI layer. Also, I didn't see a way to delete the deployment by using the deployment object itself.

>I think it is dangerous to identify the deployment to delete by its labels, as you are not 100% sure there won't have other deployment with the same labels.

I don't disagree to this, but I thought you and Tomas were specific about deleting the Deployment, which I assume to mean using the `kubeclient.Delete` function that I am currently using. So, I went ahead with fetching resource from the cluster and using it's label.

>Going this way, you would have to change DeleteDevfileComponent to be some kind of PreDeleteDevfileComponent which will run the hooks before to delete the component, but won't delete the deployment itself (as it will be deleted as part of the call to DeleteResources)

To use Delete Resources, I'd have to 1) fetch the deployment by name - this returns an `appsv1.Deployment` object,2)Convert the Deployment object into unstructured.Unstructured object with marshaling and un-marshaling - this seemed like an unnecessary task which I can't remember why.
But, if you are okay with using `DeleteResources` instead of `kubeclient.Delete(labels)`, I can make that work.